processor/src/continuation_stack.rs

```diff
@@ -0,0 +1,155 @@
+use alloc::{sync::Arc, vec::Vec};
+
+use miden_core::{
+    Program,
+    mast::{MastForest, MastNode, MastNodeId},
+};
+
+/// A hint for the initial size of the continuation stack.
+const CONTINUATION_STACK_SIZE_HINT: usize = 64;
+
+/// Represents a unit of work in the continuation stack.
+///
+/// This enum defines the different types of continuations that can be performed on MAST nodes
+/// during program execution.
+#[derive(Debug)]
+pub enum Continuation {
```

Comment:
Not for this PR: the current structure of this requires us to do a lookup into the MAST forest (via the `get_node_by_id()` function) twice for every node: first time when we push it into the continuation stack, and second time when we pop it.

Not sure if it will make practical difference, but we should be able to avoid this by removing node-specific `Start*` variants. It could look something like:
```rust
pub enum Continuation {
    StartNode(MastNodeId),
    FinishJoin(MastNodeId),
    FinishSplit(MastNodeId),
    FinishLoop(MastNodeId),
    FinishCall(MastNodeId),
    FinishDyn(MastNodeId),
    EnterForest(Arc<MastForest>),
}
```

Reply:
Ah yes good suggestion! Went for it in this PR, since it reduces the number of variants quite a bit.